Source/Machine.Fakes.FakeItEasy/FakeItEasyExpressionRewriter.cs

```diff
@@ -63,28 +63,26 @@ static Expression RewriteIsMethod(MethodCallExpression expression)
                 innerExpression);
         }
 
-        static Expression RewriteIsNullMember(MemberExpression node)
+        static Expression RewriteNullCheckingMember(MemberExpression node)
         {
             var argumentType = node.Member.DeclaringType.GetFirstTypeArgument();
             var thatAccess = GetThatAccess(argumentType);
 
-            var method = GetIsNullMethod(argumentType);
-
-            return Expression.Call(method, thatAccess);
-        }
-
-        static Expression RewriteIsNotNullMember(MemberExpression node)
-        {
-            var argumentType = node.Member.DeclaringType.GetFirstTypeArgument();
-            var thatAccess = GetThatAccess(argumentType);
-
-            var notAccess = typeof(INegatableArgumentConstraintManager<>)
-                .MakeGenericType(argumentType)
-                .MakePropertyAccess("Not", thatAccess);
+            var isValueType = argumentType.IsValueType();
+            if (isValueType)
+            {
+                // If we're checking a value type for nullity, it means the type
+                // is Nullable<T>, and we want to find IsNull<T>. (Or IsNotNull<T>.) 
+                argumentType = argumentType.GetGenericArguments().First();
+            }
 
-            var method = GetIsNullMethod(argumentType);
+            var method = typeof(ArgumentConstraintManagerExtensions)
+                .GetMethods()
+                .Where(x => x.Name == node.Member.Name)
```

Comment:
I'm taking advantage of

1. the new `IsNotNull` method, and
2. the fact that the Machine.Specs constraint names are the same as FakeItEasy's.

If it's too cutesy, I can make more explicit.